s3/download.go

```diff
@@ -0,0 +1,62 @@
+package s3util
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+type sequentialWriterAt struct {
+	w io.Writer
+}
+
+func (sw sequentialWriterAt) WriteAt(p []byte, offset int64) (int, error) {
+	return sw.w.Write(p)
```

Comment:
Won't this cause issues if concurrency is > 1?
Might want to wrap https://docs.aws.amazon.com/sdk-for-go/api/aws/#WriteAtBuffer
if so

Reply:
It's meant to be used only when concurrency = 1 .. .so private.

Reply:
I didn't really want to store the buffer contents in memory at all (could be gigs).

Reply:
oh i didn't see it was private

Reply:
=)

Reply:
Thanks for reading it carefully ^__^ much helpful